yt_dlp/extractor/tumblr.py

```diff
@@ -291,21 +440,70 @@ def _real_extract(self, url):
+            description = '\n\n'.join((
+                item.get('text') for item in content_json if item.get('type') == 'text')) or None
+        else:
+            description = self._og_search_description(webpage, default=None)
+        uploader_id = traverse_obj(post_json, 'reblogged_root_name', 'blog_name')
+
+        info_dict = {
+            'id': video_id,
+            'title': post_json.get('summary') or (blog if api_only else self._html_search_regex(
+                r'(?s)<title>(?P<title>.*?)(?: \| Tumblr)?</title>', webpage, 'title', default=blog)),
+            'description': description,
+            'uploader_id': uploader_id,
+            'uploader_url': f'https://{uploader_id}.tumblr.com/' if uploader_id else None,
+            'like_count': post_json.get('like_count'),
+            'repost_count': post_json.get('reblog_count'),
+            'age_limit': {True: 18, False: 0}.get(post_json.get('is_nsfw')),
+            'tags': post_json.get('tags'),
+        }
 
+        # for tumblr's own video hosting
+        fallback_format = None
+        formats = []
         video_url = self._og_search_video_url(webpage, default=None)
+        # for external video hosts
+        entries = []
+        ignored_providers = set()
+        unknown_providers = set()
+
+        video_jsons = (item for item in content_json if item.get('type') in ('video', 'audio'))
+        if video_jsons:
+            for video_json in video_jsons:
+                media_json = video_json.get('media') or {}
+                if api_only and not media_json.get('url') and not video_json.get('url'):
+                    raise ExtractorError('Failed to find video data for dashboard-only post')
+                provider = video_json.get('provider')
+                is_provider_ambiguous = provider in self._ambiguous_providers
+
+                if provider in ('tumblr', None):
+                    fallback_format = {
+                        'url': media_json.get('url') or video_url,
+                        'width': int_or_none(
+                            media_json.get('width') or self._og_search_property('video:width', webpage, default=None)),
+                        'height': int_or_none(
+                            media_json.get('height') or self._og_search_property('video:height', webpage, default=None)),
+                    }
+                    continue
+                elif provider in self._unsupported_providers:
+                    ignored_providers.add(provider)
+                    continue
+                elif provider and not is_provider_ambiguous and provider not in self._providers:
+                    unknown_providers.add(provider)
+
+                if video_json.get('url'):
+                    # external video host
+                    entries.append(self.url_result(
+                        video_json['url'],
+                        self._providers.get(provider, 'Generic') if not is_provider_ambiguous else None))
```

Comment:
If applying the above suggestion about moving `_ambiguous_providers` into `_providers`, we can simplify like this:
```suggestion
                elif provider and and provider not in self._providers:
                    unknown_providers.add(provider)

                if video_json.get('url'):
                    # external video host
                    entries.append(self.url_result(
                        video_json['url'], self._providers.get(provider)))
```